### PR TITLE
Share a new keyboard--Skyloong/dt40

### PR DESCRIPTION
**Pull request by JZ-Skyloong**
## Description

share a new qmk keyboard skyloong/dt40. The keymap and RGB Matrix effects are compatible with qmk and via.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

Resubmit 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/skyloong/dt40/config.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+/*
+ * RGB matrix driver config
+ */
+
+//#define WS2812_DI_PIN B4  //The pin connected to the data pin of the LEDs(define in info.json)
```
Comment (waffle87) [inline keyboards/skyloong/dt40/config.h:22]:
```suggestion
```

File: keyboards/skyloong/dt40/config.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+#define ENABLE_RGB_MATRIX_DUAL_BEACON	//Enables RGB_MATRIX_DUAL_BEACON
+#define ENABLE_RGB_MATRIX_RAINBOW_BEACON	//Enables RGB_MATRIX_RAINBOW_BEACON
+#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS	//Enables RGB_MATRIX_RAINBOW_PINWHEELS
+#define ENABLE_RGB_MATRIX_RAINDROPS	//Enables RGB_MATRIX_RAINDROPS
+#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS	//Enables RGB_MATRIX_JELLYBEAN_RAINDROPS
+#define ENABLE_RGB_MATRIX_HUE_BREATHING	//Enables RGB_MATRIX_HUE_BREATHING
+#define ENABLE_RGB_MATRIX_HUE_PENDULUM	//Enables RGB_MATRIX_HUE_PENDULUM
+#define ENABLE_RGB_MATRIX_HUE_WAVE	//Enables RGB_MATRIX_HUE_WAVE
+#define ENABLE_RGB_MATRIX_PIXEL_FRACTAL	//Enables RGB_MATRIX_PIXEL_FRACTAL
+#define ENABLE_RGB_MATRIX_PIXEL_RAIN	//Enables RGB_MATRIX_PIXEL_RAIN
```
Comment (waffle87) [inline keyboards/skyloong/dt40/config.h:55]: Ideally, these should be moved to `info.json`, and removed from this file.

File: keyboards/skyloong/dt40/dt40.c [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  { 0,  64 }, { 30,  64 }, { 45,  64 }, { 37,  64 }, { 100, 64 }, { 179, 21 }, { 140, 64 }, { 170, 64 }, { 185, 64 }, { 202, 64 },{ 224, 64 }
+
+}, {
+  // LED Index to Flag
+  4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+  4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+  4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+  4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4
+} };
+#endif
```
Comment (waffle87) [inline keyboards/skyloong/dt40/dt40.c:49]: This too should be migrated to `info.json` and removed here.

File: keyboards/skyloong/dt40/info.json [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+
+    "ws2812": {
+        "driver": "pwm",
+        "pin": "B4"
+    },
+    "rgb_matrix":{
+        "driver":"WS2812",
+        "max_brightness": 200,
+        "sleep": true
```
Comment (waffle87) [inline keyboards/skyloong/dt40/info.json:95]:
```suggestion
        "max_brightness": 200
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/skyloong/dt40/dt40.c (hunk 3 above)
Comment (JZ-Skyloong) [reply keyboards/skyloong/dt40/dt40.c:49]: I want the RGB matrix to synchronize sleep when the keyboard sleeps. Unsetting sleep in info.json, does it affect this feature?

File: keyboards/skyloong/dt40/info.json (hunk 4 above)
Comment (JZ-Skyloong) [reply keyboards/skyloong/dt40/info.json:95]: I want the RGB matrix to synchronize sleep when the keyboard sleeps. Unsetting sleep in info.json, does it affect this feature? @waffle87

File: keyboards/skyloong/dt40/config.h (hunk 2 above)
Comment (JZ-Skyloong) [reply keyboards/skyloong/dt40/config.h:50]: What is the syntax for writing RGB effects to info.json? Can you provide any guidance? I tried several methods without success. @waffle87

Comment (JZ-Skyloong) [reply keyboards/skyloong/dt40/config.h]: OK. it is done successfully.

File: keyboards/skyloong/dt40/dt40.c (hunk 3 above)
Comment (JZ-Skyloong) [reply keyboards/skyloong/dt40/dt40.c]: OK, it is migrated to info.json and removed from dt40.c .

**Review by drashna — APPROVED** [review]

**Comment by fauxpark** [comment]
There are a bunch of unrelated changes here. This PR cannot be merged as is.

**Comment by JZ-Skyloong** [comment]
> There are a bunch of unrelated changes here. This PR cannot be merged as is.

Would you please point out specific changes that are not relevant?@fauxpark

**Comment by fauxpark** [comment]
Look at the "Files changed" tab.
This happened because you merged the master branch into yours.

**Comment by JZ-Skyloong** [comment]
> Look at the "Files changed" tab. This happened because you merged the master branch into yours.

I see , thank you .

**Pull request closed without merging**